torch/nn/quantized/modules/conv.py

```diff
@@ -182,19 +183,19 @@ class Conv1d(_ConvNd):
     def __init__(self, in_channels, out_channels, kernel_size, stride=1,
                  padding=0, dilation=1, groups=1, bias=True,
                  padding_mode='zeros'):
-        kernel_size = list(_pair(kernel_size))
+        kernel_size = list(_pair_from_first(kernel_size))
```

Comment:
Why is this different than the previous `_pair` in utils.py?

Reply:
`_pair` ignorers the list inputs. `_pair_from_first` replicates the first element. Basically, if you give `kernel_size = [1]` to conv2d, _pair will ignore it, and pass it through to the aten kernel, which will throw an error. `_pair_from_first` will make sure that `1` and `[1]` are paired correctly

Reply:
I see, looks like this needs to be done this way because you are setting the `SQUEEZE_DIM` value and overriding the kernel_size. 
However the docs for conv1d say input can be int or tuple, so maybe we don't need to support list. https://pytorch.org/docs/stable/nn.html#conv1d

Also looks like quantized Conv2d actually throws an error as well when we give a list input
```
conv = torch.nn.quantized.Conv2d(5, 5, [3])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 318, in __init__
    groups, bias, padding_mode)
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 48, in __init__
    self.set_weight_bias(qweight, bias_float)
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 326, in set_weight_bias
    w, b, self.stride, self.padding, self.dilation, self.groups)
RuntimeError: Weights are expected to have 4 dimensions
```

Reply:
> I see, looks like this needs to be done this way because you are setting the `SQUEEZE_DIM` value and overriding the kernel_size.
> However the docs for conv1d say input can be int or tuple, so maybe we don't need to support list. https://pytorch.org/docs/stable/nn.html#conv1d
> 
> Also looks like quantized Conv2d actually throws an error as well when we give a list input
> 
> ```
> conv = torch.nn.quantized.Conv2d(5, 5, [3])
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 318, in __init__
>     groups, bias, padding_mode)
>   File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 48, in __init__
>     self.set_weight_bias(qweight, bias_float)
>   File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/quantized/modules/conv.py", line 326, in set_weight_bias
>     w, b, self.stride, self.padding, self.dilation, self.groups)
> RuntimeError: Weights are expected to have 4 dimensions
> ```

It doesn't matter if you give it a list or a tuple -- tuple and list in this context behave the same way. However we have to have this check, because if the arg is a tuple/list of size 1 (which is valid for conv1d), it will throw an error

Reply:
> Also looks like quantized Conv2d actually throws an error as well when we give a list input

The reason conv2d throws an error is because the size the list that you give as an arg is most probably incorrect. It has to be of `[X, Y]` or `(X, Y)` format